src/game.c

```diff
@@ -933,7 +934,19 @@ bool game_load_save(const utf8 *path)
 		return false;
 	}
 
-	bool result = game_load_sv6(rw);
+	char *extension = strrchr(path, '.');
```

Comment:
If you do use path_get_extension, it won't return NULL, it will return empty string for a extensionless file. Thus ++ will go out of bounds.